src/main/java/cz/cvut/kbss/study/rest/OidcUserController.java

```diff
@@ -56,14 +55,15 @@ public User getByUsername(@PathVariable("username") String username) {
     }
 
     @PreAuthorize(
-            "hasAuthority('" + SecurityConstants.ROLE_ADMIN + "') " +
-                    "or hasAuthority('" + SecurityConstants.ROLE_USER + "') and @securityUtils.isMemberOfInstitution(#institutionKey)")
+            "hasAuthority('" + SecurityConstants.readAllUsers + "') " +
+                    "or hasAuthority('" + SecurityConstants.readOrganizationUsers + "') and @securityUtils.isMemberOfInstitution(#institutionKey)")
     @GetMapping(produces = MediaType.APPLICATION_JSON_VALUE)
     public List<User> getUsers(@RequestParam(value = "institution", required = false) String institutionKey) {
         return institutionKey != null ? getByInstitution(institutionKey) : userService.findAll();
     }
 
-    @PreAuthorize("hasAuthority('" + SecurityConstants.ROLE_ADMIN + "') or #username == authentication.name")
+    @PreAuthorize("hasAuthority('" + SecurityConstants.writeAllUsers + "') or #username == authentication.name or" +
+            " hasAuthority('" + SecurityConstants.writeOrganizationUsers + "') and @securityUtils.areFromSameInstitution(#username)")
```

Comment:
```suggestion
    @PreAuthorize("hasAuthority('" + SecurityConstants.writeAllUsers + "') or " +
            "hasAuthority('" + SecurityConstants.writeOrganizationUsers + "') and @securityUtils.areFromSameInstitution(#username) or " + 
            "#username == authentication.name")
```

Reply:
@palagdan I would like to be somehow consistent in the way we specify security conditions within `@PreAuthorize` ... 

We should either specify it from the most specific condition to the most general condition or the opposite way around. I like specific => general, but within ~90% of code you did it already general => specific, which is fine with me as well. 

This is the reason why I rewrote this `@PreAuthorize` the same way as you did it elsewhere. Please check it elsewhere as well ...